lib/topological_inventory.rb

```diff
@@ -2,7 +2,7 @@
 class TopologicalInventory
   DEFAULT_PATH_PREFIX = 'api'.freeze
   SERVICE_NAME = 'topological-inventory'.freeze
-  VERSION = 'v0.1'.freeze
+  VERSION = 'v1.0'.freeze
```

Comment:
How do we feel about specifying just the major version and allow the api to return the latest minor as it should be backwards compatible?

In our discussion on code promotion I would expect the Topology api to add a new revision about once every sprint (every 2 weeks) or more.

@syncrou thoughts?

Reply:
Yes I agree.  Specifying the major version will allow us to always use the newest minor version in use by the generated client.

Reply:
@gmcculloug @syncrou 
When only the major version is specified Topology Service sends back a redirect.
```
[----] D, [2019-05-18T04:50:47.524256 #9456:3fd7a9b9b9a8] DEBUG -- : ETHON: performed EASY effective_url=https://x.x.x.x/api/topological-inventory/v1/service_offerings/7     response_code=302 return_code=ok total_time=1.3775089999999999
```
With other services like RBAC we don't get a redirect.
The typheous library that Openapi generator uses for client code has a follow redirects but it seems not be exposed in the config for openapi. 
Also trying to see if the IPP mentions auto redirects on the server side or should a redirect be send back to the client

Reply:
The IPP says
```
$VERSION may be a full major.minor version (eg., /v1.5/), or just the major version (eg,. /v1/) of the API you wish to use. If you only specify a major version, the application should serve the latest minor version available. Please see the Versioning section below for more discussion.
```

Reply:
This is the only API where we have embedded the version in our code as opposed to just using the version that the embedded Gem has defined in it.

Reply:
@mkanoor 
This issue was opened to work on the redirect: https://github.com/ManageIQ/manageiq-api-common/issues/66

Yes, this is the only API that we have embedded the version, and as we have discussed this in the past and I still feel that the project should continue to have some control over the versions used for this and extend it to others clients.

If the gems were versioned we would limit this in the `Gemfile` with something like `~> 1.0` where we would pick up newer versions but not include any `v2` or above.

Since we do not have a versioned gem at the moment we are relying on defining it here and use the API versioning to keep us within the backwards compatible v1 APIs.

Obviously we would want to use `v1` here and drop the `.0`, but we can probably wait until the redirect issue is resolved.

cc @Fryguy

Reply:
I'm not too familiar with this repo, but I don't understand the problem that you're trying to solve.  Your `Gemfile` should specify something like:
```ruby
gem "topological_inventory-api-client", "~>1.0"
```

Since the API client should be published every time the API is published, this would allow you to use the latest `1.Y` version, set a minimum `Y` value and avoid getting 2.Y of the API before you're ready for it.  You could also add something like "< 1.5" if you were not yet ready for a new attribute, method or behavior.  Using v1.0 of the client against v1.5 of the API (which `/api/v1` will eventually redirect to) could return unexpected results.  I'm not sure how the client would handle extra attributes in a response payload, so I would advise against overriding the base path defined in the `openapi.json`.

Reply:
My concern with using the major version is whether the generated gem can handle when "extra" data might come back that it doesn't expect.  That being said, I also agree with @bdunne that you if we release clients with every version bump, then you can use the client as built and not even have to worry about this at all.

Reply:
After working through how bundler reads `version.rb` on a gem and running a number of tests locally. I recommend we modify the `Gemfile` to:
```
gem 'topological_inventory-api-client', '~> 1.0', :git => "https://github.com/ManageIQ/topological_inventory-api-client-ruby"
```
Since this change achieves what the `base_path` override was doing, I suggest removing the `base_path` modifications from the client library.

tl;dr
I realize that's what @bdunne @Fryguy are suggesting, I didn't fully understand the process `bundle` uses with `version.rb` in a non published gem. 
I was under the impression before that a `git tag` controlled what version was pulled, and since we had none on this gem, I thought the `:branch` key was the only option.

Reply:
We should release v1.0 of this gem and remove the `:git` reference as well.